modules/compute/gke-node-pool/main.tf

```diff
@@ -162,8 +160,8 @@ resource "google_container_node_pool" "node_pool" {
 
     reservation_affinity {
       consume_reservation_type = var.reservation_affinity.consume_reservation_type
-      key                      = local.specific_reservations_count != 1 ? null : local.reservation_resource_api_label
-      values                   = local.specific_reservations_count != 1 ? null : [for reservation in var.reservation_affinity.specific_reservations : reservation.name]
+      key                      = local.input_specific_reservations_count != 1 ? null : local.reservation_resource_api_label
+      values                   = local.input_specific_reservations_count != 1 ? null : [for reservation in var.reservation_affinity.specific_reservations : reservation.name]
```

Comment:
use `local. verified_specific_reservations` instead so and pass in the full qualified name of the reservation so that shared reservations work.

```
values = length(local.verified_specific_reservations) != 1 ? null : [for reservation in local.verified_specific_reservations : reservation.id]
```